### PR TITLE
chore(143): e2e tests for aggregate targets

### PR DESCRIPTION
**Pull request by ralfudx**
# Description

[#143](https://github.com/medic/care-teams/issues/143)

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:e2e-tests-for-aggregate-targets/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:e2e-tests-for-aggregate-targets/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:e2e-tests-for-aggregate-targets/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@ralfudx to just show your changes, update the PR's base to `9231-add-aggregate-filter`
Screenshot 2024-07-23 at 9 57 53 AM

**Comment by ralfudx** [comment]
@latin-panda @lorerod @tatilepizs I have opened this PR and added you(s) as reviewers. Kindly have a look when you have some time. Thanks

**Review by tatilepizs — CHANGES_REQUESTED** [review]
Thank you for adding the e2e tests to cover this new feature ⭐ 

I left some suggestions, let me know if you have any questions 🙂

File: tests/e2e/default/targets/target-aggregates.wdio-spec.js [hunk 1]
```diff
@@ -347,6 +360,30 @@ describe('Target aggregates', () => {
       await targetAggregatesPage.expectTargetDetails(target);
     });
 
+    it('should filter aggregates by place', async () => {
```
Comment (lorerod) [inline tests/e2e/default/targets/target-aggregates.wdio-spec.js:363]:
```suggestion
    it('should filter aggregates by place when user has multiple facilities', async () => {
```

Comment (lorerod) [inline tests/e2e/default/targets/target-aggregates.wdio-spec.js:363]: Can you also add steps for this scenario to change and validate the selected place?

File: tests/e2e/default/targets/target-aggregates.wdio-spec.js [hunk 2]
```diff
@@ -235,82 +323,7 @@ describe('Target aggregates', () => {
     it('should display correct data', async () => {
```
Comment (lorerod) [inline tests/e2e/default/targets/target-aggregates.wdio-spec.js:323]: Can you confirm that the side filter doesn't show if a user has only one facility? Does it make sense to add a validation in this scenario or create a new one to test this specifically? Let me know what you think.
cc: @latin-panda

**Review by lorerod — CHANGES_REQUESTED** [review]
Thanks @ralfudx ! I just left a couple of suggestions inline. Let me know what you think.

File: tests/page-objects/default/targets/target-aggregates.wdio.page.js [hunk 3]
```diff
@@ -122,6 +125,19 @@ const clickOnTargetAggregateListItem = async (contactId) => {
...
+    await (await sidebarFilterBtn()).click();
+  }
+  return await (await filterCloseBtn()).waitForDisplayed();
+};
+
+const selectFilterOption = async () => {
+  if (await (await filterCloseBtn()).isDisplayed()) {
+    await (await filterRadioBtn()).click();
+  }
+};
```
[the remark this answers is not in the document]
Comment (ralfudx) [reply tests/page-objects/default/targets/target-aggregates.wdio.page.js:139]: Thanks for the feedback @tatilepizs they sound good to me and I really appreciate your help with this

File: tests/e2e/default/targets/target-aggregates.wdio-spec.js (hunk 2 above)
Comment (latin-panda) [reply tests/e2e/default/targets/target-aggregates.wdio-spec.js:323]: Yes, the sidebar should not appear if the user has only 1 facility assigned. So, it's better to have a scenario for the user with 1 facility and another scenario for the user with 2 or 3 facilities.

File: tests/page-objects/default/analytics/analytics.wdio.page.js [hunk 4]
```diff
@@ -4,6 +4,8 @@ const TARGET_UNMET_COLOR = '#000000';
 
 const goToTargets = () => browser.url('/#/analytics/targets');
 
+const goToTargetAggregates = () => browser.url('/#/analytics/target-aggregates');
```
Comment (latin-panda) [inline tests/page-objects/default/analytics/analytics.wdio.page.js:7]: We already have a function to navigate [to aggregate targets, here. ](https://github.com/medic/cht-core/blob/4e73a79c71c5e37d892e975052d4bb2a8961e7e8/tests/page-objects/default/targets/target-aggregates.wdio.page.js#L26) 
Use that one because it handles loaders for you already. Then, revert the changes in this file.

File: tests/e2e/default/targets/target-aggregates.wdio-spec.js
Comment (latin-panda) [inline tests/e2e/default/targets/target-aggregates.wdio-spec.js:1]: Check for the facility indicator ([see screenshots)](https://github.com/medic/cht-core/pull/9282) when a user has multiple facilities assigned.

1. it should default to the first option of the filter. The indicator shows the facility name.
2. it should update the indicator with a newly selected facility filter (different from the default)

**Review by latin-panda — CHANGES_REQUESTED** [review]
@ralfudx Please find my comments below

**Comment by tatilepizs** [comment]
Closed because the changes from the branch `filter-aggregate-period` where needed to complete the tests. This work was completed on [chore(143): e2e tests for aggregate targets (place, period)](https://github.com/medic/cht-core/pull/9305)
